### ISSUE TITLE
tabdown.lexer: Linebreaks are not properly assigned

### ISSUE THREAD

**Issue by freshdried** [freshdried/tabdown#1]
#### How do we assign newlines?
```
1
    2

3
```
correct:
`1 { 2 } \n 3`

incorrect
`1 { 2 \n } 3`

---

```
1
    2

    3
```
correct:
`1 { 2 \n 3 }`

incorrect
`1 { 2 } \n { 3 }`

---

Maybe we need a buffer to hold newlines
b/c their membership is ambiguous until the next element.
If there is no next element (i.e. end of source), we ignore the trailing newlines, because they are indefinitely ambiguous.

**Comment by freshdried** [comment]
### Solution:


#### input
```
1
    2
        3


    4
```
#### algorithm:
```
1 { 2 { 3 
1 { 2 { 3                ( n )
1 { 2 { 3                ( n, n )
1 { 2 { 3                ( n, n )  {, 4 
1 { 2 { 3                ( n, n )  {, 4 
1 { 2 { 3 }              ( n, n )  4
1 { 2 { 3 } n n          4
1 { 2 { 3 } n n 4
```
For each run (each character), store linebreaks in an additional intermediary array.
Linebreaks are then pushed after dedents. If there are no dedents, we just push linebreaks. If there is an additional indents, (there is therefore no dedents) we push linebreaks, then push indent.

**Comment by freshdried** [comment]
Fixed by 7f939719fef16f5ab4535352a3d8dcc0fafa9445
